project/main.py

```diff
@@ -88,8 +84,7 @@ def run(experiment: Experiment) -> tuple[str, float | None, dict]:
     )
 
     metric_name, error, metrics = evaluation(experiment)
-    if wandb.run:
-        wandb.finish()
```

Comment:
Make sure to add the `wandb.finish` somewhere though

Reply:
I believe lines 60-68 contain the new wandb run logic - I may very well be missing something